### PR TITLE
feat: Helm chart secrets-store-csi-driver-provider-aws

### LINKED ISSUE #1306: Why was `csi-secrets-store-provider-aws` namespace changed to `kube-system`?
- [X ] ✋ I have searched the open/closed issues and my issue is not listed.

#### Please describe your question here

Recently `csi-secrets-store-provider-aws` namespace has ben updated from `csi-secrets-store-provider-aws` to `kube-system`
I believe it was done under the auspices of https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1172
I do not understand why.
Our cluster, that was created using `v4.17.0`, now fails to update using `v4.20.0` with the following:
```
│ Error: namespaces "kube-system" already exists
│ 
│   with module.cluster_addons.module.csi_secrets_store_provider_aws[0].kubernetes_namespace_v1.csi_secrets_store_provider_aws,
│   on .terraform/modules/cluster_addons/modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf line 6, in resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws":
│    6: resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws" {
```
The `csi-secrets-store-provider-aws` namespace is removed and then the apply fails; leaving us without any `csi-secrets-store-provider-aws`!

Is there anyway to either continue using the old `csi-secrets-store-provider-aws` namespace or; if `kube-system` is the preferred namespace, to fix that error and make the update succeed in moving `csi-secrets-store-provider-aws` to `kube-system`?

A bit concerned that this change seems to have broken backwards compatibility.
 
#### Provide a link to the example/module related to the question
https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/modules/kubernetes-addons/csi-secrets-store-provider-aws

### PR DESCRIPTION
**Pull request by atheiman**
### What does this PR do?

- Replaces https://github.com/aws/eks-charts/tree/master/stable/csi-secrets-store-provider-aws in favor of https://github.com/aws/secrets-store-csi-driver-provider-aws/tree/main/charts/secrets-store-csi-driver-provider-aws.
- adds a check to the addon's namespace creation to not create kube-system.

### Motivation

- Resolves #1306 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

This is now working in my own eks cluster with the default `secrets-store-csi-driver` addon (`enable_secrets_store_csi_driver = true`). I have not thoroughly tested - I only have this working currently w/ no helm_config overrides passed to the addon module. Probably should be tested with namespace = "kube-system" for example.

### CODE REVIEW

**Comment by atheiman** [comment]
@balakrishnasoura-okta I tried locally with the updated main.tf on this branch and still get this error. Something in the helm charts must have changed? secretproviderclasses-admin-role is now created in chart secrets-store-csi-driver and the second chart csi-secrets-store-provider-aws is still trying to create it? Sorry im pretty new to most of this so not great w/ debugging it.

```
module.tf_eks.module.csi_secrets_store_provider_aws.module.helm_addon.helm_release.addon[0]: Creating...
╷
│ Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "secretproviderclasses-admin-role" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "csi-secrets-store-provider-aws": current value is "secrets-store-csi-driver"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "kube-system": current value is "secrets-store-csi-driver"
│
│   with module.tf_eks.module.csi_secrets_store_provider_aws.module.helm_addon.helm_release.addon[0],
│   on .terraform/modules/tf_eks.csi_secrets_store_provider_aws.helm_addon/modules/kubernetes-addons/helm-addon/main.tf line 1, in resource "helm_release" "addon":
│    1: resource "helm_release" "addon" {
```

**Comment by balakrishnasoura-okta** [comment]
I will troubleshoot by EOD today and update it in the PR

**Comment by atheiman** [comment]
I have this working in an eks cluster now w/ no overrides passed to the addon module that is installing. I replaced the now-deprecated eks-charts/csi-secrets-store-provider-aws with github.com/aws/secrets-store-csi-driver-provider-aws. Probably needs more testing to be sure of this change, and not sure what else you want to update related to swapping out the chart.

**Comment by egarbi** [comment]
> Details

It seems the Title needs to be adjusted to show a `version` can you @atheiman do this to speed up the process? We are waiting for this to be fixed. Many thanks for the quick action

**Comment by egarbi** [comment]
pre-commit output showing the following
Screenshot 2023-02-03 at 10 12 20

File: modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws" {
+  count = try(var.helm_config["create_namespace"], true) && local.namespace != "kube-system" ? 1 : 0
   metadata {
     name = local.namespace
   }
 }
 
 module "helm_addon" {
-  source = "../helm-addon"
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints/modules/kubernetes-addons/helm-addon"
```
Comment (egarbi) [inline modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf:14]: You should revert this to be a local path
```suggestion
  source = "../helm-addon"
```

**Comment by mchlbataller** [comment]
Hello guys, any updates so far? Waiting for this to be merged 🤞

**Comment by dannykruitbosch** [comment]
Is there a work-around for this issue. We really need this fixed.

**Comment by atheiman** [comment]
Im not sure if they plan on merging this change or what still needs to be done b/c i think in some cases this could be a breaking change. I actually ended up not using the secretsmanager secret provider. But this workaround worked for me for a few days.

Most of the flags to enable features simply trigger additional sub modules of this tf project to be run. If you have any issue w/ how any of the submodules are configured, just copy them out of this project and run them as your own submodule that depends on your blueprints module call. So for this issue you could copy the secrets mgr secret store provider tf submodule out of my branch and put it in your codebase. Each submodule takes a long list of standard arguments about the cluster, most/all are available as outputs from the eks cluster module.

Unfortunately i dont have this workaround in place anymore so I dont have an example I can copy to show you, but hopefully this gets you down a path to a reasonable workaround.

**Comment by bryantbiggs** [comment]
thank you for the PR - this will be resolved in the new addons module https://github.com/aws-ia/terraform-aws-eks-blueprints-addons

**Pull request closed without merging**